code/game/objects/structures/ai_core.dm

```diff
@@ -58,11 +59,20 @@
 		update_appearance()
 
 /obj/structure/ai_core/Destroy()
+	if(istype(remote_ai))
+		remote_ai.break_core_link()
+		remote_ai = null
 	QDEL_NULL(circuit)
 	QDEL_NULL(core_mmi)
 	QDEL_NULL(laws)
 	return ..()
 
+/obj/structure/ai_core/take_damage(damage_amount, damage_type, damage_flag, sound_effect, attack_dir, armour_penetration)
+	if(istype(remote_ai))
+		to_chat(remote_ai, span_danger("Your core is under attack!"))
+	. = ..()
```

Comment:
(IMO) You should respect the parent return call when handling this message
```suggestion
	. = ..()
	if(. > 0 && istype(remote_ai))
		to_chat(remote_ai, span_danger("Your core is under attack!"))
```
`.` is the amount of damage done, so we can make this only send the message if damage was done